src/drivers/sifive,spi0.c

```diff
@@ -122,43 +140,81 @@ int __metal_driver_sifive_spi0_transfer(struct metal_spi *gspi,
                                       char *tx_buf,
                                       char *rx_buf)
 {
+    /* Transmit and receive buffer must not be empty */
+    if (tx_buf == NULL || rx_buf == NULL) {
+        return 1;
+    }
+
     struct __metal_driver_sifive_spi0 *spi = (void *)gspi;
     int rc = 0;
-    int rxdata = 0;
 
     rc = configure_spi(spi, config);
     if(rc != 0) {
         return rc;
     }
 
     /* Hold the chip select line for all len transferred */
-    METAL_SPI_REGW(METAL_SPI_REG_CSMODE) = METAL_SPI_CSMODE_HOLD;
-
-    for(int i = 0; i < len; i++) {
-        /* Wait for TXFIFO to not be full */
-        while(METAL_SPI_REGW(METAL_SPI_REG_TXDATA) & METAL_SPI_TXDATA_FULL) ;
+    METAL_SPI_REGW(METAL_SPI_REG_CSMODE) &= ~(METAL_SPI_CSMODE_MASK);
+    METAL_SPI_REGW(METAL_SPI_REG_CSMODE) |= METAL_SPI_CSMODE_HOLD;
 
-        /* Transfer byte */
-        if(tx_buf) {
-            METAL_SPI_REGW(METAL_SPI_REG_TXDATA) = tx_buf[i];
```

Comment:
Did simply writing the byte not work? It shouldn't be necessary to make writing `txdata` more complicated.

Maybe you can replace `METAL_SPI_REGW()` with`METAL_SPI_REGB()`?

Reply:
`METAL_SPI_REGB` works